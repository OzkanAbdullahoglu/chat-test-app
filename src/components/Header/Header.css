.header-root {
    display: flex;
    align-items: center;
    height: 48px;
    position: fixed;
    width: 100%;
    max-width: 480px;
    justify-content: flex-start;
    background-color: #effcff;
    box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    -moz-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    -o-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
}

.container-root {
    width: 100%;
    height: 48px;
    display: flex;
    position: fixed;
    justify-content: center;
}

.incoming-user {
    height: 72px;
    background-color: #fffbf0;
    top: 48px;
}

.header-content {
    position: relative;
    margin-left: auto;
    margin-right: auto;
    display: flex;
}

.header-text {
    font-size: 1rem;
    font-family: "BrownRegular", "Helvetica", "Arial", sans-serif;
    font-weight: 400;
}