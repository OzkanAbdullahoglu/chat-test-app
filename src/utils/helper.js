/* eslint-disable linebreak-style */
import React, { useState, useEffect } from 'react';
export const timeConverter = (timeStamp) => {
  if (!timeConverter.cache) {
    timeConverter.cache = {};
  }

  if (!timeConverter.cache[timeStamp]) {
    const timeStampMs = timeStamp * 1000;
    const dateObj = new Date(timeStampMs);
    const dayOfMonth = dateObj.getDate() < 10 ? `0${dateObj.getDate()}` : dateObj.getDate();
    const months = [
      'January', 'February', 'March', 'April', 'May', 'June',
      'July', 'August', 'September', 'October', 'November', 'December',
    ];
    const curMonth = months[dateObj.getMonth()];
    const curYear = dateObj.getFullYear();
    const curMinute = dateObj.getMinutes() < 10 ? `0${dateObj.getMinutes()}` : dateObj.getMinutes();
    const curHour = (dateObj.getHours() < 10 ? `0${dateObj.getHours()}` : dateObj.getHours());

    timeConverter.cache[timeStamp] = {
      date: `${curMonth} ${dayOfMonth},${curYear}`,
      time: `${curHour}:${curMinute}`,
    };
    return timeConverter.cache[timeStamp];
  }
  return timeConverter.cache[timeStamp];
};

export const isToday = (timestamp, date) =>{
  const today = Math.floor(Date.now() / 1000).toString();
  console.log(today);
  return timeConverter(today).date === timeConverter(timestamp).date ?
    'Today' : timeConverter(today).date === timeConverter(timestamp).date - 1 ?
    'Yesterday' : date;
};




function getCurrentViewDims() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useCurrentViewDims() {
  const [windowDimensions, setWindowDimensions] = useState(getCurrentViewDims());

  useEffect(() => {
    function handleResize() {
      setWindowDimensions(getCurrentViewDims());
    }

    window.addEventListener('scroll', handleResize);
    return () => window.removeEventListener('scroll', handleResize);
  }, []);

  return windowDimensions;
}

