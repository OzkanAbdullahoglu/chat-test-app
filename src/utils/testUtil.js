/* eslint-disable linebreak-style */
import { createStore, applyMiddleware } from 'redux';
import { middlewares } from '../store';
import rootReducer from '../reducers';

const chat = {
  requestedData: [
    {
      id: 1,
      direction: 'in',
      status: 'read',
      timestamp: '1577834102',
      text: 'Alice was beginning',
    },
  ],
  isPending: false,
  error: '',
  scrollDownEnable: true,
};

const message = {
  messagesEndRef: '',
};

const version = { number: '0.0.1' };

export const initialStates = { chat, message, version };

export const storeFactory = (initalState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initalState);
};
