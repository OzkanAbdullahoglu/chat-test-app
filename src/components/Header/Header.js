/* eslint-disable linebreak-style */
import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import PropTypes from 'prop-types';
import './Header.css';
import {
  chatActions, getUnreadMessages,
} from '../../reducers';
import { Wrapper } from '../Wrapper/Wrapper';
import SvgIcon from '../Icon/Icon';

const Header = ({
  unreadMessages,
  incomingIcon,
  incomingAvatar,
}) => (
  <div className="container-root">
    {incomingAvatar ? (
      <div className="header-root incoming-user">
        <Wrapper
          data-test="wrapper"
          incomingIcon
          aria-label="Back to chat screen"
          tabIndex="0"
        >
          <SvgIcon
            incomingIcon
            viewBox="0 0 48 48"
          >
            <path xmlns="http://www.w3.org/2000/svg" d="M20 12l-2.83 2.83 9.17 9.17-9.17 9.17 2.83 2.83 12-12z" />
            <path xmlns="http://www.w3.org/2000/svg" d="M0 0h48v48h-48z" fill="none" />
          </SvgIcon>
        </Wrapper>
        <Wrapper
          incomingAvatar
          aria-label="View User 2 profile"
          tabIndex="0"
        >
          <SvgIcon
            incomingAvatar
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </SvgIcon>
        </Wrapper>
        <div className="header-content" tabIndex="0">
          <span
            className="header-text"
          >User 2</span>
        </div>
      </div>
    ) : (
      <div className="header-root" >
        <Wrapper
          data-test="wrapper"
          incomingAvatar={false}
          aria-label="View User 1 profile"
          tabIndex="0"
        >
          <SvgIcon
            incomingAvatar={false}
            viewBox="0 0 24 24"
            data-test="icon"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </SvgIcon>
        </Wrapper>
        <div className="header-content" tabIndex="0">
          <span
            className="header-text"
          >User 1 ({unreadMessages.count} new messages)</span>
        </div>
      </div>
    )
    }
  </div>
);

Header.propTypes = {
  unreadMessages: PropTypes.object,
  incomingIcon: PropTypes.bool,
  incomingAvatar: PropTypes.bool,
};

const mapStateToProps = (store) => ({
  unreadMessages: getUnreadMessages(store),
});

const withRedux = connect(
  mapStateToProps,
  { ...chatActions }
);

export default compose(withRedux)(Header);
