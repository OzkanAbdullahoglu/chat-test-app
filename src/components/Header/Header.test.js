/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from './Header';
import { storeFactory } from '../../utils/testUtil';

const defaultProps = {
  incomingAvatar: false,
  unreadMessages: {
    count: 0,
    firstUnreadId: 5,
  } };
const setup = () => {
  const store = storeFactory();
  return shallow(<Header store={store} {...defaultProps} />).dive().dive();
};

describe('Header Component', () => {
  const header = setup();
  it('renders without crashing', () => {
    expect(toJson(header)).toMatchSnapshot();
  });
  it('contains Wrapper component', () => {
    expect(header.find('[data-test="wrapper"]').exists()).toBe(true);
  });
  it('contains SvgIcon component', () => {
    expect(header.find('[data-test="icon"]').exists()).toBe(true);
  });
});
describe('Header Component', () => {
  const header = setup();
  const { unreadMessages } = defaultProps;
  it('renders properly up to `incomingAvatar` and `unreadMessages` prop', () => {
    expect(header.find('[className="header-text"]').text()).toBe(`User 1 (${unreadMessages.count} new messages)`);
  });
  it('renders conditionally up to `incomingAvatar`', () => {
    header.setProps({
      incomingAvatar: true,
      unreadMessages: {
        count: 18,
        firstUnreadId: 5,
      } });
    expect(header.find('[className="header-text"]').text()).toBe('User 2');
  });
});
