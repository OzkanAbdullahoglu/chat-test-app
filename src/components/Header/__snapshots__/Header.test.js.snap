// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders without crashing 1`] = `
<div
  className="container-root"
>
  <div
    className="header-root"
  >
    <styled.div
      aria-label="View User 1 profile"
      data-test="wrapper"
      incomingAvatar={false}
      tabIndex="0"
    >
      <styled.svg
        data-test="icon"
        incomingAvatar={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </styled.svg>
    </styled.div>
    <div
      className="header-content"
      tabIndex="0"
    >
      <span
        className="header-text"
      >
        User 1 (
        0
         new messages)
      </span>
    </div>
  </div>
</div>
`;
